osquery/remote/http_client.cpp

```diff
@@ -181,11 +281,21 @@ void Client::encryptConnection() {
                                client_options_.sni_hostname_->c_str());
   }
 
-  boost_system::error_code rc;
-  ssl_sock_->handshake(boost_asio::ssl::stream_base::client, rc);
+  ssl_sock_->async_handshake(
```

Comment:
@uptycs-nishant, it looks like this async request does not have a timer-wait started before it.

I see the others have:
```
    if (client_options_.timeout_) {
      timer_.async_wait(
          [=](boost_system::error_code const& ec) { timeoutHandler(ec); });
    }
```

Is there a reason for skipping `async_handshake` this?